rest-service/manager_rest/test/endpoints/test_plugins_updates.py

```diff
@@ -0,0 +1,365 @@
+                'Requested `PluginsUpdate` with ID `non_existing` was '
+                'not found') as e:
+            self.client.plugins_update.get('non_existing')
+            self.assertEqual(404, e.exception.status_code)
+
+    @suppress_plugins_change_detection
+    def test_returns_correct_plugins_update(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd1')
+        self.wait_for_deployment_creation(self.client, 'd1')
+        plugins_update = self.client.plugins_update.update_plugins(
+            'hello_world')
+        response_plugins_update = self.client.plugins_update.get(
+            plugins_update.id)
+        self.assertEqual(plugins_update.id, response_plugins_update.id)
+
+
+@attr(client_min_version=base_test.LATEST_API_VERSION,
+      client_max_version=base_test.LATEST_API_VERSION)
+class PluginsUpdateTest(BaseServerTestCase):
+    """
+    Test plugins update.
+    """
+
+    def setUp(self):
+        super(PluginsUpdateTest, self).setUp()
+        self._sm = get_storage_manager()
+
+    def test_raises_with_nonexisting_blueprint(self):
+        with self.assertRaises(CloudifyClientError) as e:
+            self.client.plugins_update.update_plugins("non_existing")
+            self.assertEqual(404, e.exception.status_code)
+
+    @suppress_plugins_change_detection
+    def test_raises_blueprint_has_no_deployments(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        with self.assertRaisesRegexp(CloudifyClientError,
+                                     "The blueprint hello_world has no "
+                                     "deployments to update."):
+            self.client.plugins_update.update_plugins("hello_world")
+
+    @suppress_plugins_change_detection
+    def test_plugins_update_and_execution_parameters_are_correct(self):
+        self.put_file(*self.put_blueprint_args(blueprint_id='hello_world'))
+        self.client.deployments.create('hello_world', 'd1')
+        self.wait_for_deployment_creation(self.client, 'd1')
+        self.client.deployments.create('hello_world', 'd2')
+        self.wait_for_deployment_creation(self.client, 'd2')
+        plugins_update = self.client.plugins_update.update_plugins(
+            'hello_world')
+        self.assertListEqual(['d1', 'd2'],
+                             plugins_update.deployments_to_update)
+        execution = self.client.executions.get(plugins_update.execution_id)
+        self.assertDictEqual(
+            execution.parameters,
+            {'update_id': plugins_update.id,
+             'deployments_to_update': ['d1', 'd2'],
+             'temp_blueprint_id': plugins_update.temp_blueprint_id})
+
+    @suppress_plugins_change_detection
+    def test_plugins_manager_is_a_singleton(self):
+        self.assertTrue(
+            get_plugins_updates_manager() == get_plugins_updates_manager())
+
+    @suppress_plugins_change_detection
```

Comment:
Actually, it seems you can put this on the entire class, you can turn the patch on and off in the class setup teardown